rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -282,6 +360,92 @@ trait NodeInfoServiceCached extends NodeInfoService  with Loggable with CachedRe
   }
 
   /**
+   * An utility method that gets data from backend for things that are
+   * node really nodes (pending or deleted).
+   */
+  private[this] def getNotAcceptedNodeDataFromBackend(status: InventoryStatus): Box[Map[NodeId, NodeInfo]] = {
```

Comment:
Ok ... I tried like 15 minutes, I see it's complicated and it's ok it's two seprated methods that looks a lot the same :)